### PR TITLE
draft of annotations blog

### PR DESCRIPTION
**Pull request by michellevballen**
I'm not sure if the images are set up properly. Would love feedback on the content. Thanks!

### CODE REVIEW

File: content/post/adding_context_with_annotations.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+title: "Adding Context to Your Analysis with Annotations"
+cover: ""
+publishDate: "2018-07-31"
+draft: true
+tags:
+ - documentation
+ - annotations
+---
+
+Anyone who has worked in digital analytics will tell you that day over day performance can be volatile. Shifts in marketing mix can cause fluctuating e-commerce conversion rates, new feature launches can lead to sudden and temporary swings in engagement rates and onsite bugs can result in anomalies in abandonment rates. Some of these scenarios can be diagnosed through extensive segmentation of data. Others, [like a dropped analytics snippet] (https://www.locallyoptimistic.com/post/web-analytics-process/) or a bug with your payment processor, cannot be so easily uncovered. The simplest thing to do when events like these take place is to take a mental note and count on your memory for when you inevitably have to revisit that data in the future. As you can imagine, taking a mental note isn’t a scalable solution for a number of reasons.
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Suggestion:

Change 

>As you can imagine, taking a mental note isn’t a scalable solution for a number of reasons.

to

>Unfortunately, taking a mental note isn’t a scalable solution.

For increased emphasis.

File: content/post/adding_context_with_annotations.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+It may seem like an ‘unsexy’ practice relative to other data processes, but from my experience, keeping a record of the online and offline events that affect your business is a practice and well worth the (small) time investment.
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Suggestion:

> While it's not the most _thrilling_ task for a data team, keeping a record of the online and offline events that affect your business is a practice and well worth the (small) time investment.

Additionally, some people find using the term "sexy" for professional descriptions to be misogynistic, so I'd prefer we at least strike that :)

File: content/post/adding_context_with_annotations.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+<!--more-->
+
+#### Why taking a mental note isn’t a scalable solution:
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Suggestion:

Change the section heading to

> #### Examples of Common Failures

And then rephrase the bullets to emphasize the failure point (your second bullet is really good here).

File: content/post/adding_context_with_annotations.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+* At one point or another, a company executive will most certainly probe you to diagnose an anomaly or shift in performance (read: a dip in performance) -- if you can’t recall all of the events that occurred around the start of the shift there is a good chance that you’ll find yourself going down a rabbit hole try to figure it out.
+* As your company grows, there will undoubtedly be new members joining the analytics team. New analysts can’t possibly be expected to remember major events that they weren’t there to experience. Similarly, veteran employees who were there to experience major events leave the organization at one point or another.
+
+In this post I will walk through the steps I take to integrate annotations into my data models and visualizations. I will refer to an imaginary home goods e-commerce company called “Utensell” to guide the walkthrough.
+
+### Step 1: Whenever something notable happens, record it in a google spreadsheet. Easy!
+
+[screenshot_1]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-google-sheet.jpg
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: `Example of Event Tracking Spreadsheet`

File: content/post/adding_context_with_annotations.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+Examples of things I keep note of:
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: > Things to Track:

File: content/post/adding_context_with_annotations.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+* Product Launches (e.g., launched new silverware sku)
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: s/sku/SKU

File: content/post/adding_context_with_annotations.md (hunk 5 above)
Comment (mikekaminsky) [reply content/post/adding_context_with_annotations.md]: Also you need an additional carriage return to get your bullets to format correctly.

File: content/post/adding_context_with_annotations.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+* Fraud (e.g., surge in bot traffic)
+
+Depending on the business, it may make sense to keep note of some additional metadata. In this example, I have a boolean flag to indicate whether there was a discount promotion associated with the event. I also keep track of links that are relevant to the notable event (e.g., links to press articles) in a separate column.
+
+_Daily Time Spent: I generally spend 1-5 minutes per day on this step._
+
+
+### Step 2: Automate the formatting and uploading of your annotations log to your database.
+
+I use the [datasheets](https://github.com/Squarespace/datasheets) python library (created by the team at Squarespace) to reformat and upload my annotations log table into Utensell’s data warehouse. But you can use any method you prefer! [Quick shoutout to Rujuta Kortikar, who built V1 of this script using the gspread library before datasheets was released].
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: These square brackets are doing weird things to the markdown. I don't think this needs to be a parenthetical statement, so I'd just drop the brackets.

File: content/post/adding_context_with_annotations.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+[screenshot_2]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-jupyter-notebook.jpg
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: See correct mardkown example above, and please use a better description for the image (for accessibility).

File: content/post/adding_context_with_annotations.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+### Step 3: Reference your annotations log whenever and wherever you’d like!
+
+Let’s imagine that I want to run an analysis that looks at daily orders data for the past 7 days. Knowing that discount promotions lead to big spikes in order volume, I may want to incorporate that information into my analysis. One way to achieve this would be to join my annotations log table to my order facts table. Example below -
+
+```
+SELECT
+ orders.created_at AS order_created_date,
+ CASE WHEN annotations.discount_promotion_flag IS TRUE
+   THEN 'Discount Promotion' ELSE 'No Promotion' END AS discount_promotion_flag,
+ count(DISTINCT orders.id) AS total_orders
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: This is such a bike-shed comment, but _my_ preferred SQL formatting syntax for this statement would look like:

```
CASE 
   WHEN annotations.discount_promotion_flag IS TRUE
       THEN 'Discount Promotion' 
   ELSE 'No Promotion' 
END AS discount_promotion_flag,
COUNT(DISTINCT orders.id) AS total_orders
```

But this is a Michael style rule, not a LO style rule so feel free to leave as is :)

File: content/post/adding_context_with_annotations.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+FROM utensell_order_facts AS orders
+ LEFT JOIN utensell_annotations_log AS annotations
+   ON annotations.date = date(orders.created_at)
+WHERE orders.created_at >= dateadd(day, -6, getdate()) AND orders.created_at <= getdate()
+GROUP BY 1,2
+ORDER BY 1;
+```
+
+The output for this query would look something like this -
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Prefer colon over dash here.

File: content/post/adding_context_with_annotations.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+| 7/20/18  | No Promotion | 135 |
+| 7/21/18  | No Promotion | 127 |
+| 7/22/18  | No Promotion | 173 |
+| 7/23/18  | Discount Promotion | 346 |
+
+Now I can easily see a correlation between order volume and discount promotions. With that knowledge, I may decide to add a where clause that excludes dates when a discount promotion ran.
+
+As another example, imagine that I have a daily dashboard in my business intelligence tool (in this case, Looker) that gets emailed company-wide and one of the charts looks like this -
+
+[screenshot_3]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-graph-before.jpg
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Same as above (repeat for all images)

File: content/post/adding_context_with_annotations.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+In this visualization, I immediately see some unexplainable ups and downs in the data. I can already predict the question my teammates will ask when they see this data -- “what happened around the 25th of June that caused such a big spike in conversion rate?”.
+
+With the annotations log data, I can now get ahead of their question and add that layer of context onto the visualization. One option for achieving this would be to pivot on the ‘is promotional date’ dimension --
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Let's use a full stop or a colon to end these sentences leading into images.

File: content/post/adding_context_with_annotations.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+Or simply add an asterisk to the x-axis labels where promotions were running --
+
+[screenshot_5]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-graph-after-v2.jpg
+
+Finally, I might just want to simply include a table with notable dates directly at the bottom of a dashboard. That way, when business users notice data anomalies, they can quickly cross reference a list of notable dates to diagnose the cause.
+
+[screenshot_6]: https://github.com/locallyoptimistic/LocallyOptimistic/blob/master/static/img/annotations-notable-dates-table.jpg
+
+Some challenges to consider before adopting this process:
```
Comment (mikekaminsky) [inline content/post/adding_context_with_annotations.md]: Carriage return needed.

File: content/post/adding_context_with_annotations.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+
+While it's not the most _thrilling_ task for a data team, keeping a record of the online and offline events that affect your business is a practice that is well worth the (small) time investment.
+
+<!--more-->
+
+#### Examples of Common Failures:
+* You are tasked with evaluating the success of a recent marketing campaign and decide to generate a benchmark for comparison using historical data. You fail to account for a major initiative that took place during your 'pre' period and you circulate an evaluation and recommended next steps that are not sound.
+* A company executive sees a graph displaying month over month e-commerce conversion rate and probes you diagnose a shift in performance (read: a dip in performance). You can’t recall all of the events that occurred around the start of the dip and you find yourself going down a rabbit hole trying to make sense of the shift in performance.
```
Comment (sbreitenother) [inline content/post/adding_context_with_annotations.md]: "probes you" -> "asks you to"

File: content/post/adding_context_with_annotations.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+ - annotations
+---
+
+Anyone who has worked in digital analytics will tell you that day over day performance can be volatile. Shifts in marketing mix can cause fluctuating e-commerce conversion rates, new feature launches can lead to sudden and temporary swings in engagement rates and onsite bugs can result in anomalies in abandonment rates. Some of these scenarios can be diagnosed through extensive segmentation of data. Others, [like a dropped analytics snippet](https://www.locallyoptimistic.com/post/web-analytics-process/) or a bug with your payment processor, cannot be so easily uncovered. The simplest thing to do when events like these take place is to take a mental note and count on your memory for when you inevitably have to revisit that data in the future. Unfortunately, taking a mental note isn’t a scalable solution.
+
+While it's not the most _thrilling_ task for a data team, keeping a record of the online and offline events that affect your business is a practice that is well worth the (small) time investment.
+
+<!--more-->
+
+#### Examples of Common Failures:
```
Comment (sbreitenother) [inline content/post/adding_context_with_annotations.md]: You use sentence case for other headings

File: content/post/adding_context_with_annotations.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+* You are tasked with evaluating the success of a recent marketing campaign and decide to generate a benchmark for comparison using historical data. You fail to account for a major initiative that took place during your 'pre' period and you circulate an evaluation and recommended next steps that are not sound.
+* A company executive sees a graph displaying month over month e-commerce conversion rate and probes you diagnose a shift in performance (read: a dip in performance). You can’t recall all of the events that occurred around the start of the dip and you find yourself going down a rabbit hole trying to make sense of the shift in performance.
+* A new member joins the analytics team and gets started on their first analysis. They can't be expected to remember major events that they weren’t there to experience so they ultimately send out a flawed analysis. Similarly, a veteran employee who _was_ there to experience major events leaves the organization without sharing their domain knowledge with the rest of the team.
```
Comment (sbreitenother) [inline content/post/adding_context_with_annotations.md]: 'analytics team' -> 'Analytics team'

File: content/post/adding_context_with_annotations.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+title: "Adding Context to Your Analysis with Annotations"
+cover: ""
+publishDate: "2018-07-31"
+draft: true
+tags:
+ - documentation
+ - annotations
+---
+
+Anyone who has worked in digital analytics will tell you that day over day performance can be volatile. Shifts in marketing mix can cause fluctuating e-commerce conversion rates, new feature launches can lead to sudden and temporary swings in engagement rates and onsite bugs can result in anomalies in abandonment rates. Some of these scenarios can be diagnosed through extensive segmentation of data. Others, [like a dropped analytics snippet](https://www.locallyoptimistic.com/post/web-analytics-process/) or a bug with your payment processor, cannot be so easily uncovered. The simplest thing to do when events like these take place is to take a mental note and count on your memory for when you inevitably have to revisit that data in the future. Unfortunately, taking a mental note isn’t a scalable solution.
```
Comment (sbreitenother) [inline content/post/adding_context_with_annotations.md]: `[like a dropped analytics snippet](https://www.locallyoptimistic.com/post/web-analytics-process/) ` -> `[like a dropped analytics snippet]({{< relref "web-analytics-process.md" >}})`

File: content/post/adding_context_with_annotations.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+Finally, I might just want to simply include a table with notable dates directly at the bottom of a dashboard. That way, when business users notice data anomalies, they can quickly cross reference a list of notable dates to diagnose the cause.
+
+![Notable Dates Table](img/annotations-notable-dates-table.jpg)
+
+Some challenges to consider before adopting this process:
+
+* Compared to other data best practices, keeping an annotations log is a pretty manual process that requires you stay plugged into all of the areas of the business
+* If the annotations log has multiple contributors (vs. one owner) you may run into some inconsistencies with nomenclature and / or notable events that were missed completely or added twice
+
+Sometimes numbers speak for themselves. Other times, they just need a little context.
```
Comment (sbreitenother) [inline content/post/adding_context_with_annotations.md]: Consider removing "just" - might make it more powerful

**Comment by sbreitenother** [comment]
Looks good. Merging in draft mode. Will make separate PR to remove draft and add cover image (once we have compressed version of illustration). Also will check images are displayed correctly before removing draft.

**Pull request merged**